### PR TITLE
add workflow for syncing

### PR DESCRIPTION
**Pull request by gshennvm**
# What does this PR do ?

Add a one line overview of what this PR aims to accomplish.

dev should be branch that tracks main but has all the commits whenever nemo changes things. We will sync this branch with main every release

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -4,14 +4,18 @@ Thanks for the interest in contributing to NeMo-Aligner. We do all of NeMo-Align
...
-**Send your PRs to the `main` branch**
+**Send your PRs to the `main` or `dev` branch**
 
 1) Make sure your PR does one thing. Have a clear answer to "What does this PR do?".
 2) Read General Principles and style guide below
 3) Make sure you sign your commits. E.g. use ``git commit -sS`` when committing.
 4) Make sure all unittests finish successfully before sending PR ``pytest`` or (if your dev box does not have GPU) ``pytest --cpu`` from the root folder
 5) Send your PR and request a review
 
+**NOTE**: The `main` branch uses a fixed nemo version which we will update on every release. The `dev` branch is the branch that has all the commits from aligner main but is up to date with nemo main, this branch is less stable but we run nightly tests on this branch to make sure everything works. We only provide the dockerfile that works with `main`.
```
Comment (odelalleau) [inline CONTRIBUTING.md:15]: Some small suggestions to try to make it even clearer:
```suggestion
**NOTE**: The `main` branch uses a fixed NeMo version which we will update on every release. The `dev` branch is the branch that has all commits from `main` but uses NeMo's main branch: this branch is less stable but we run nightly tests on it to make sure everything works. We only provide the dockerfile that works with `main`, which is the branch most PRs should target unless they require the latest NeMo main (in which case they should target `dev`).
```

**Review by odelalleau — COMMENTED** [review]
I'm not familiar enough with workflows to be able to review the yaml config, but at first glance it seems reasonable.

**Comment by gshennvm** [comment]
> I'm not familiar enough with workflows to be able to review the yaml config, but at first glance it seems reasonable.

thanks! I think it should be okay, I got it from github market place. It should work well

File: nemo_aligner/models/nlp/gpt/megatron_gpt_reward_model.py [hunk 2]
```diff
@@ -318,7 +318,7 @@ def on_load_checkpoint(self, checkpoint) -> None:
         """
         # mcore uses distributed checkpointing
         if "state_dict" in checkpoint and checkpoint["state_dict"]:
-            for index, module in enumerate(self.get_gpt_module_list()):
```
Comment (gshennvm) [inline nemo_aligner/models/nlp/gpt/megatron_gpt_reward_model.py:321]: whoops: will remove once everything else looks good to you

File: nemo_aligner/utils/train_utils.py [hunk 3]
```diff
@@ -38,7 +38,7 @@ def set_sync_funcs(ptl_model, forward_only):
         param_sync_func = ptl_model.sync_overlap_parameters
 
     # pipeline schedules will get these from ptl_model.model.config
-    for module in ptl_model.get_gpt_module_list():
```
Comment (gshennvm) [inline nemo_aligner/utils/train_utils.py:41]: whoops again: will remove once everything else looks good to you

**Review by odelalleau — APPROVED** [review]
Good to merge after removing the `get_model_module_list()` calls

**Pull request closed without merging**